localstack/services/s3/stream_patch.py

```diff
@@ -0,0 +1,532 @@
+        value: bytes | IO[bytes],
+        storage: Optional[str] = None,
+        etag: Optional[str] = None,
+        multipart: Optional[StreamedFakeMultipart] = None,
+        encryption: Optional[str] = None,
+        kms_key_id: Optional[str] = None,
+        bucket_key_enabled: Any = None,
+        lock_mode: Optional[str] = None,
+        lock_legal_status: Optional[str] = None,
+        lock_until: Optional[str] = None,
+        checksum_value: Optional[str] = None,
+    ) -> StreamedFakeKey:
+        key_name = clean_key_name(key_name)
+        if storage is not None and storage not in s3_models.STORAGE_CLASS:
+            raise s3_exceptions.InvalidStorageClass(storage=storage)
+
+        bucket = self.get_bucket(bucket_name)
+
+        # getting default config from bucket if not included in put request
+        if bucket.encryption:
+            bucket_key_enabled = bucket_key_enabled or bucket.encryption["Rule"].get(
+                "BucketKeyEnabled", False
+            )
+            kms_key_id = kms_key_id or bucket.encryption["Rule"][
+                "ApplyServerSideEncryptionByDefault"
+            ].get("KMSMasterKeyID")
+            encryption = (
+                encryption
+                or bucket.encryption["Rule"]["ApplyServerSideEncryptionByDefault"]["SSEAlgorithm"]
+            )
+        if isinstance(value, bytes):
+            value = BytesIO(value)
+
+        new_key = StreamedFakeKey(
+            name=key_name,
+            bucket_name=bucket_name,
+            value=value,
+            account_id=self.account_id,
+            storage=storage,
+            etag=etag,
+            is_versioned=bucket.is_versioned,
+            # AWS uses VersionId=null in both requests and responses
+            version_id=str(random.uuid4()) if bucket.is_versioned else "null",
+            multipart=multipart,
+            encryption=encryption,
+            kms_key_id=kms_key_id,
+            bucket_key_enabled=bucket_key_enabled,
+            lock_mode=lock_mode,
+            lock_legal_status=lock_legal_status,
+            lock_until=lock_until,
+            checksum_value=checksum_value,
+        )
+
+        existing_key = bucket.keys.get(key_name)
+        if existing_key:
+            existing_key.is_latest = False
+
+        bucket.keys[key_name] = new_key
+
+        return new_key
+
+    @patch(s3_models.S3Backend.list_object_versions, pass_target=False)
+    def list_object_versions(
+        self,
+        bucket_name: str,
+        delimiter: Optional[str] = None,
+        key_marker: Optional[str] = None,
+        prefix: str = "",
+    ) -> Tuple[list[StreamedFakeKey], list[str], list[s3_models.FakeDeleteMarker]]:
+        bucket = self.get_bucket(bucket_name)
+
+        common_prefixes: list[str] = []
+        requested_versions: list[StreamedFakeKey] = []
+        delete_markers: list[s3_models.FakeDeleteMarker] = []
+        all_versions = list(itertools.chain(*(l for key, l in bucket.keys.iterlists())))
```

Comment:
this needs to be patched, because it uses `copy.deepcopy` which uses pickle under the hood, and fails to pickle the temp file. I've changed the logic, because deep copying it was basically dumping every file to disk to load them back with the patching done in moto.